src/hotspot/share/opto/type.cpp

```diff
@@ -4613,6 +4613,11 @@ const TypePtr *TypeAryPtr::with_instance_id(int instance_id) const {
   return make(_ptr, _const_oop, _ary->remove_speculative()->is_ary(), _klass, _klass_is_exact, _offset, instance_id, _speculative, _inline_depth);
 }
 
+const TypePtr *TypeAryPtr::with_offset(int offset) const {
```

Comment:
I think you can use the existing
https://github.com/openjdk/jdk/blob/49b86224aacc7fd8b4d3354a85d72ef636a18a12/src/hotspot/share/opto/type.cpp#L4892